python/taichi/misc/gui.py

```diff
@@ -53,7 +53,10 @@ def set_image(self, img):
       img = img.to_numpy()
     assert isinstance(img, np.ndarray)
     assert len(img.shape) in [2, 3]
-    img = img.astype(np.float32)
+    if img.dtype not in [np.float32, np.float64]:
+      img = img.astype(np.float32) * (1.0 / 255)
+    else:
+      img = img.astype(np.float32)
```

Comment:
Do you think this should be here, or in `cook_image`? Or add `set_image_raw(u8 *img)` in pybind11?

Reply:
You mean we should invoke `cook_image` in `GUI.set_image`? I think that's a great idea!

Reply:
Yeah it is. But I mean, is `GUI.set_image` expected to support `np.uint8` format image?

Reply:
Yes :-)